docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+full status for each deployed object when the workload is delivered only to one 
+cluster. The controller watches *appliedmanifestworks* objects to determine which 
+objects have been delivered through the work agent. It then starts dynamic informers 
+to watch those objects, collect their individual statuses, and report back the status 
+updating *workstatuses* objects in the namespace associated with the WEC in the ITS.
+
+## KubeStellar Controllers Architecture
+
+The KubeStellar controllers architecture is based on common patterns and best 
+practices for Kubernetes controllers, such as the 
+[<u>Kubernetes Sample Controller</u>](https://github.com/kubernetes/sample-controller). 
+A Kubernetes controller uses informers to watch for changes in Kubernetes
+objects, caches to store the objects, event handlers to react to
+events, work queues for parallel processing of tasks, and a reconciler
+to ensure the actual state matches the desired state. However, that
+pattern has been extended to provide the following features:
+
+- Using dynamic informers
+- Starting informers on all API Resources (except some that do not need
+  watching)
+- Informers and Listers references are maintained in a hash map and
+  indexed by GVK (Group, Version, Kind) of the watched objects.
+- Using a common work queue and set of workers, where the key is defined as follows:
+  - Key is a struct instead than a string, and contains the following:
+    - GVK of the informer and lister for the object that generated the
+      event
+    - Structured Name of the object 
+    - For delete event: Shallow copy of the object being deleted. This
+      is required for objects w/o finalizers that need to be deleted
+      from the managed clusters (WECs)
+- Starting & stopping informers dynamically based on creation or
+  deletion of CRDs (which add/remove APIs on the WDS).
+- One client connected to the WDS space and one (or more in the future)
+  to connect to one or more OCM shards.
+  - The WDS-connected client is used to start the dynamic
+    informers/listers for most API resources in the WDS
+  - The OCM-connected client is used to start informers/listers for OCM
+    ManagedClusters and to copy/update/remove the wrapped objects
+    into/from the OCM mailbox namespaces.
+
+Currently there are two controllers in the KubeStellar controller
+manager: the placement controller and the status controllor.
+
+### Placement Controller
+
+The Placement controller is responsible for watching workload and 
+placement objects, and wrapping and delivering objects to the ITS
+based on placement policies.
+
+The architecture and the event flow for create/update object events is
+illustrated in Figure 3 (some details might be omitted to make the flow easier
+to understand).
+
+<figure>
+  <img src="assets/placement-controller.png"  alt="Placement Controller">
+  <figcaption align="center">Figure 3 - Placement Controller</figcaption>
+</figure>
```

Comment:
That picture says the informers have no handlers for deletion events. That is untrue, if I recall correctly.

Reply:
That is correct, thanks, I fixed the picture to show the deletion event as well for the handler.